azul/src/display_list.rs

```diff
@@ -1752,6 +1754,16 @@ fn push_background(
                     None => TypedSize2D::new(image_dimensions.0, image_dimensions.1)
                 };
 
+                if let Some(repeat) = background_repeat {
+                    let info = match repeat {
+                        StyleBackgroundRepeat::NoRepeat => LayoutPrimitiveInfo::with_clip_rect(info.rect, TypedRect::from_size(size))
+                    };
+
+                    push_image(&info, builder, app_resources, image_id, size);
+                } else {
+                    push_image(info, builder, app_resources, image_id, size);
+                };
+
                 push_image(info, builder, app_resources, image_id, size);
```

Comment:
Why is `push_image` here twice, seems like a bug?

You could also simplify a bit with:

```rust
use StyleBackgroundRepeat::*;
let background_size = match background_repeat {
     Some(NoRepeat) => LayoutPrimitiveInfo::with_clip_rect(info.rect, TypedRect::from_size(size)),
     None = *info,
};

push_image(&background_size, builder, app_resources, image_id, size);
```